client/state/selectors/README.md

```diff
@@ -9,5 +9,49 @@ When adding a new selector to this directory, make note of the following details
 
 - Each new selector exists in its own file, named with [kebab case](https://en.wikipedia.org/wiki/Kebab_case) (dash-delimited lowercase)
 - There should be no more than a single default exported function per selector file
+- The JSDoc header for the default exported function should contain an alias tag with the name of the function in it and that should match the camelCase conversion of the filename itself
```

Comment:
@dmsnell I think since it's [documented](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#selectors) that a default export of a file in the `state/selectors` folder is assumed to be a selector, the tag would be unnecessary noise. We could even generate `index.js` as-is, if we so choose. Despite the challenges, I think we should try to identify the default export by parsing the file.

Roughly, I think we could check the type of `declaration` when visiting the `ExportDefaultDeclaration` node and, if an identifier, check all top-level `body` children on the `Program:exit` visitor to find the matching identifier (optionally repeating if the matched identifier is itself a variable). When found we can pull the comments from the node's `leadingComents` property.

https://astexplorer.net/#/QdklGbkjVm